### PR TITLE
Add documentation for the base64_decode filter

### PR DESCRIPTION
**Pull request by paulusbrand**
## Proposed change

This PR adds documentation for the base64_decode filter.
The linked pull request on the codebase describes the addition of a optional argument for this filter.

## Type of change

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [ ] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [x] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logos and icons in [Brands repository](https://github.com/home-assistant/brands).
- [ ] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information

- Link to parent pull request in the codebase: [PR](https://github.com/home-assistant/core/pull/116603)
- Link to parent pull request in the Brands repository: 
- This PR fixes or closes issue: fixes #

## Checklist

- [x] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [x] The documentation follows the Home Assistant documentation [standards].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

## Summary by CodeRabbit

- **New Features**
  - Introduced new base64 decoding filters in the templating functionality.
  - Added filter `base64_decode` for decoding with default utf-8 encoding.
  - Added filter `base64_decode("ascii")` for decoding with ascii encoding.
  - Added filter `base64_decode(None)` for decoding to raw bytes.

### CODE REVIEW

**Review by edenhaus — DISMISSED** [review]
Please update the PR to have the latest changes from the core PR

File: source/_docs/configuration/templating.markdown [hunk 1]
```diff
@@ -1121,6 +1121,20 @@ Some examples:
...
+- Filter `value | base64_decode` Decodes a base 64 string to a string, by default utf-8 encoding is used.
+- Filter `value | base64_decode("ascii")` Decodes a base 64 string to a string, using ascii encoding.
+- Filter `value | base64_decode(None)` Decodes a base 64 string to raw bytes.
+
+<div class='note'>
+
+Some examples:
+{% raw %}
+- `{{ "aG9tZWFzc2lzdGFudA==" | base64_decode }}` - renders as `homeassistant`
+- `{{ "aG9tZWFzc2lzdGFudA==" | base64_decode("raw") }}` - renders as `b'homeassistant'`
```
Comment (frenck) [inline source/_docs/configuration/templating.markdown]: As per request above by @edenhaus, please update the documentation to match the upstream core PR.

**Review by frenck — CHANGES_REQUESTED** [review]

File: source/_docs/configuration/templating.markdown (hunk 1 above)
Comment (paulusbrand) [reply source/_docs/configuration/templating.markdown]: Sorry, I was not paying attention. Done

**Review by frenck — APPROVED** [review]
Thanks, @paulusbrand 👍 

../Frenck

**Pull request merged**